src/TorchSharp/Utils/tensorboard/Summary.cs

```diff
@@ -0,0 +1,127 @@
+
+                    /// <summary>
+                    /// Outputs a `Summary` protocol buffer with images.
+                    /// The summary has up to `max_images` summary values containing images. The
+                    /// images are built from `tensor` which must be 3-D with shape `[height, width,
+                    /// channels]` and where `channels` can be:
+                    /// *  1: `tensor` is interpreted as Grayscale.
+                    /// *  3: `tensor` is interpreted as RGB.
+                    /// *  4: `tensor` is interpreted as RGBA.
+                    /// The `name` in the outputted Summary.Value protobufs is generated based on the
+                    /// name, with a suffix depending on the max_outputs setting:
+                    /// *  If `max_outputs` is 1, the summary value tag is '*name*/image'.
+                    /// *  If `max_outputs` is greater than 1, the summary value tags are
+                    ///    generated sequentially as '*name*/image/0', '*name*/image/1', etc.
+                    /// </summary>
+                    /// <param name="tag"> A name for the generated node. Will also serve as a series name in TensorBoard. </param>
+                    /// <param name="tensor">
+                    /// A 3-D `uint8` or `float32` `Tensor` of shape `[height, width,
+                    /// channels]` where `channels` is 1, 3, or 4.
+                    /// 'tensor' can either have values in [0, 1] (float32) or [0, 255] (uint8).
+                    /// The image() function will scale the image values to [0, 255] by applying
+                    /// a scale factor of either 1 (uint8) or 255 (float32). Out-of-range values
+                    /// will be clipped.
+                    /// </param>
+                    /// <param name="rescale"></param>
+                    /// <param name="dataformats"></param>
+                    /// <returns> A scalar `Tensor` of type `string`. The serialized `Summary` protocol buffer. </returns>
+                    public static Tensorboard.Summary image(string tag, torch.Tensor tensor, double rescale = 1, string dataformats = "NCHW")
+                    {
+                        tensor = torch.utils.tensorboard.utils.convert_to_HWC(tensor, dataformats);
+                        int scale_factor = calc_scale_factor(tensor);
+                        tensor = tensor.to_type(torch.ScalarType.Float32);
+                        tensor = (tensor * scale_factor).clip(0, 255).to_type(torch.ScalarType.Int8);
+                        Tensorboard.Summary.Types.Image image = make_image(tensor, rescale);
+                        var summary = new Tensorboard.Summary();
+                        summary.Value.Add(new Tensorboard.Summary.Types.Value() { Tag = tag, Image = image });
+                        return summary;
+                    }
+
+                    public static Tensorboard.Summary image(string tag, string file_name, double rescale = 1)
```

Comment:
Add doc comments to each of the overloads.

Reply:
Done